include/mbedtls/cipher.h

```diff
@@ -672,10 +698,11 @@ int mbedtls_cipher_finish( mbedtls_cipher_context_t *ctx,
 /**
  * \brief               This function writes a tag for AEAD ciphers.
  *                      Currently supported with GCM and ChaCha20+Poly1305.
- *                      Must be called after mbedtls_cipher_finish().
+ *                      This must be called after mbedtls_cipher_finish().
  *
- * \param ctx           The generic cipher context.
- * \param tag           The buffer to write the tag to.
+ * \param ctx           The generic cipher context. This must be initialized.
+ * \param tag           The buffer to write the tag to. This must be a readable
+ *                      boffer of at least \p tag_len Bytes.
  * \param tag_len       The length of the tag to write.
```

Comment:
In line with other changes, I think we should accept `tag == NULL` if `taglen == 0`.